pkg/testutils/kustainer/kustainer.go

```diff
@@ -128,6 +135,56 @@ func (c *KustainerContainer) PortForward(ctx context.Context, config *rest.Confi
 	return fmt.Errorf("failed to connect to kustainer after retries: %w", lastErr)
 }
 
+// monitorConnection periodically checks the endpoint and reconnects if needed.
+func (c *KustainerContainer) monitorConnection(ctx context.Context, config *rest.Config, podName string) {
```

Comment:
This feels like a lot ... I'm pretty surprised we're dropping simple connections on a single VM. (I do see them in the logs.)

I'd really prefer we turn on more telemetry to find out why this is dropping - maybe Kustainer needs more resources/is crashing/something else? We can beef up the size of the runners https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/about-larger-runners but I think it's worth trying to get some telemetry first for why this seems to happen so much.

I can try to help with this in the near future.